ignite/pkg/events/events.go

```diff
@@ -74,25 +91,53 @@ func (e Event) Text() string {
 }
 
 // Bus is a send/receive event bus.
-type Bus chan Event
+type (
+	Bus struct {
+		evchan chan Event
+		buswg  *sync.WaitGroup
+	}
+
+	BusOption func(*Bus)
+)
+
+func WithWaitGroup(wg *sync.WaitGroup) BusOption {
+	return func(bus *Bus) {
+		bus.buswg = wg
+	}
+}
 
 // NewBus creates a new event bus to send/receive events.
-func NewBus() Bus {
-	return make(Bus)
+func NewBus(options ...BusOption) *Bus {
+	bus := &Bus{
+		evchan: make(chan Event),
+	}
+
+	for _, apply := range options {
+		apply(bus)
+	}
+
+	return bus
 }
 
 // Send sends a new event to bus.
-func (b Bus) Send(e Event) {
-	if b == nil {
+func (b *Bus) Send(e Event) {
+	if b == nil || b.evchan == nil {
 		return
 	}
-	b <- e
+	if b.buswg != nil {
+		b.buswg.Add(1)
+	}
+	b.evchan <- e
+}
+
+func (b *Bus) Events() <-chan Event {
```

Comment:
```suggestion
```

Reply:
The goal of the function is to protect the channel from outside interaction. The encapsulated channel should be wrote only through `Send()`, read only through `Events()` and close only with `Shutdown()`. If this channel became a public field it now could be wrote directly and it is not so secure. We still can add ability to tweak buffer size from outside.

Reply:
We decided the receive events channel from outside right? In this case this func can be removed, because channel is already exposed.

Reply:
There is an idea to make it publicly available But event bus now is not just a channel, it is a channel with associated wait group and there is a logic of adding units to the wait group on receiving events. So having full access to the channel from outside gives an ability to send data to channel directly bypassing `Send()`. And what are the cases where we could want to manage channel directly. I think they would be we could make channel public this in the future.

Reply:
The use case would be to allow passing a channel with a custom buffer size. 

We could init it internally, but we should implement a new option like WithCustomBufferSize.